environments/common/src/main/java/org/jboss/weld/environment/deployment/discovery/jandex/JandexClassFileInfo.java

```diff
@@ -291,9 +292,34 @@ private boolean containsAnnotation(ClassInfo classInfo, DotName requiredAnnotati
                 return true;
             }
         }
+        // Also check default methods on interfaces
+        for (DotName interfaceName : classInfo.interfaceNames()) {
+            final ClassInfo interfaceInfo = index.getClassByName(interfaceName);
+            if (interfaceInfo == null) {
+                // we are accessing a class that is outside of the jandex index
+                // fallback to using reflection
+                Class<?> interfaceClass = loadClass(interfaceName.toString());
+                for (Method method : interfaceClass.getDeclaredMethods()) {
+                    if (method.isDefault() && method.isAnnotationPresent(requiredAnnotation)) {
+                        return true;
+                    }
+                }
+                continue;
+            }
+            for (MethodInfo method : interfaceInfo.methods()) {
+                if (isDefault(method) && method.hasAnnotation(requiredAnnotationName)) {
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
+    private boolean isDefault(MethodInfo method) {
```

Comment:
This method should either check whether the method is declared on an interface or it should be renamed to something more general such as `isPublic()`. In future it could be leveraged elsewhere in which case someone might forget to only invoke it on interface methods.

Reply:
Good point. On the other hand this method is private... `isNonAbstractPublicInstanceMethod()`?

Reply:
+1